src/coreclr/jit/optimizer.cpp

```diff
@@ -7118,36 +7098,38 @@ void Compiler::optHoistLoopBlocks(unsigned loopNum, ArrayStack<BasicBlock*>* blo
                 Value& top = m_valueStack.TopRef();
                 assert(top.Node() == stmt->GetRootNode());
 
+                // hoist the top node?
                 if (top.m_hoistable)
                 {
                     m_compiler->optHoistCandidate(stmt->GetRootNode(), block, m_loopNum, m_hoistContext);
                 }
                 else
                 {
                     JITDUMP("      [%06u] not %s: %s\n", dspTreeID(top.Node()),
-                            top.m_invariant ? "invariant" : "hoistable", top.m_failReason);
+                            top.m_invariant ? "hoistable" : "invariant", top.m_failReason);
```

Comment:
Just to avoid confusion in future as to why we would print `"hoistable"` for a condition of `m_invariant == true`, can you write it like this (basically have `"not"` written next to the text and add a comment stating the reason.

```suggestion
                            JITDUMP("      [%06u] %s: %s\n", dspTreeID(top.Node()), top.m_invariant ? "not hoistable" : "not invariant", top.m_failReason);
```

Reply:
Done.